src/assets/PugAsset.js

```diff
@@ -0,0 +1,124 @@
+const Asset = require('../Asset');
+const HTMLAsset = require('./HTMLAsset');
+const localRequire = require('../utils/localRequire');
+
+// A list of all attributes that may produce a dependency
+// Based on https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes
+const ATTRS = {
+  src: [
+    'script',
+    'img',
+    'audio',
+    'video',
+    'source',
+    'track',
+    'iframe',
+    'embed'
+  ],
+  href: ['link', 'a', 'use'],
+  srcset: ['img', 'source'],
+  poster: ['video'],
+  'xlink:href': ['use'],
+  content: ['meta']
+};
+
+// A regex to detect if a variable is a 'pure' string (no evaluation needed)
+const PURE_STRING_REGEX = /(^"([^"]+)"$)|(^'([^']+)'$)/g;
+
+class PugAsset extends Asset {
+  constructor(name, pkg, options) {
+    super(name, pkg, options);
+    this.type = 'html';
+  }
+
+  async parse(code) {
+    const load = await localRequire('pug-load', this.name);
+    const lexer = await localRequire('pug-lexer', this.name);
+    const parser = await localRequire('pug-parser', this.name);
+    const linker = await localRequire('pug-linker', this.name);
+    const filters = await localRequire('pug-filters', this.name);
```

Comment:
that's a lot of packages... I wonder if there is a way we can just require the `pug` package rather than all of these separately.

Reply:
Those packages needed for handling internal dependencies, such `include` and `extend`. Calling `pug` onto source file will not be reload those files at changing. That's why we need parse AST and do other things.

Reply:
hmm, it seems like `pug.compile` already returns a list of dependencies. Can we just use that rather than re-implementing the ast walking ourselves? https://github.com/pugjs/pug/blob/master/packages/pug/lib/index.js#L192

Reply:
@devongovett Ok, I'll check it